pkgs/by-name/libresoc/ieee754fpu.nix

```diff
@@ -0,0 +1,77 @@
+{
+  python39Packages,
+  fetchFromLibresoc,
+  bigfloat,
+  sfpy,
+  symbiyosys,
+  nmutil,
+  nmigen,
+  pytest-output-to-files,
+}:
+with python39Packages;
+  buildPythonPackage rec {
+    name = "libresoc-ieee754fpu";
+    pname = "ieee754fpu";
+    version = "unstable-2024-03-31";
+
+    src = fetchFromLibresoc {
+      inherit pname;
+      hash = "sha256-Ghbvg2Y4YlmxVEa3EtcvEVai4hC4VU4q+XIQh4pQ7+c=";
+      rev = "829dfbc53ba38ec17bc544cb0b862e73cee223db"; # HEAD @ version date
+    };
+
+    prePatch = ''
+      touch ./src/ieee754/part{,_ass,_cat,_repl}/__init__.py
+    '';
+
+    propagatedBuildInputs = [nmutil];
+
+    nativeCheckInputs = [pytestCheckHook pytest-xdist pytest-output-to-files nmigen symbiyosys sfpy bigfloat];
```

Comment:
You're mixing arguments and elements of `python39Packages`, this is hard to understand. Please split into two lists, one that uses arguments, and another one that is `(with python39Packages; [ ... ])`.